src/System.Management.Automation/engine/hostifaces/PowerShellProcessInstance.cs

```diff
@@ -54,16 +54,32 @@ public PowerShellProcessInstance(Version powerShellVersion, PSCredential credent
         {
             string exePath = PwshExePath;
             string processArguments = string.Empty;
+            bool startingWindowsPowerShell51 = false;
 
 #if !UNIX
             // if requested PS version was "5.1" then we start Windows PS instead of PS Core
             // Version parameter needs to go before all other parameters.
-            if (powerShellVersion != null)
+            startingWindowsPowerShell51 = (powerShellVersion != null) && (powerShellVersion.Major == 5) && (powerShellVersion.Minor == 1);
+            if (startingWindowsPowerShell51)
             {
-                if ((powerShellVersion.Major == 5) && (powerShellVersion.Minor == 1))
+                exePath = WinPwshExePath;
+                processArguments = "-Version 5.1";
+
+                if (useWow64)
                 {
-                    exePath = WinPwshExePath;
-                    processArguments = "-Version 5.1";
+                    string procArch = Environment.GetEnvironmentVariable("PROCESSOR_ARCHITECTURE");
+
+                    if ((!string.IsNullOrEmpty(procArch)) && (procArch.Equals("amd64", StringComparison.OrdinalIgnoreCase) ||
+                        procArch.Equals("ia64", StringComparison.OrdinalIgnoreCase)))
+                    {
+                        exePath = WinPwshExePath.ToLowerInvariant().Replace("\\system32\\", "\\syswow64\\");
```

Comment:
Perhaps using `internal static ProcessorArchitecture GetProcessorArchitecture()` would be more right.
The method will be fixed in #11046

Reply:
This arch-detection code was a copy-paste from the same location in Windows PS code;
But I think it is better to change this to use GetProcessorArchitecture().
Updated.

Reply:
Is `GetProcessorArchitecture()` getting the processor-architecture process-wide, or system-wide? We want the process-wide value here.

Reply:
Very good point. As far as i can tell:
* currently `GetProcessorArchitecture()` returns system-wide info (and. btw, [comment on this function ](https://github.com/PowerShell/PowerShell/blob/78d64664ceda59d4a6918be1842f16da38d68c7d/src/System.Management.Automation/utils/PsUtils.cs#L139) is wrong).
* after changes of  #11046 `GetProcessorArchitecture()` will return architecture of the current process.